clients/hadoopfs/src/main/java/io/lakefs/LakeFSFileSystem.java

```diff
@@ -264,29 +273,32 @@ public boolean mkdirs(Path path, FsPermission fsPermission) throws IOException {
     public FileStatus getFileStatus(Path path) throws IOException {
         LOG.debug("$$$$$$$$$$$$$$$$$$$$$$$$$$$$ getFileStatus, path: {} $$$$$$$$$$$$$$$$$$$$$$$$$$$$ ", path.toString());
         ObjectLocation objectLoc = pathToObjectLocation(path);
-        if (objectLoc == null) {
-            throw new FileNotFoundException(path.toString());
-        }
         try {
             ObjectsApi objectsApi = new ObjectsApi(this.apiClient);
             ObjectStats objectStat = objectsApi.statObject(objectLoc.getRepository(), objectLoc.getRef(), objectLoc.getPath());
-            long length = 0;
-            Long sizeBytes = objectStat.getSizeBytes();
-            if (sizeBytes != null) {
-                length = sizeBytes;
-            }
-            long modificationTime = 0;
-            Long mtime = objectStat.getMtime();
-            if (mtime != null) {
-                modificationTime = TimeUnit.SECONDS.toMillis(mtime);
-            }
-            Path filePath = path.makeQualified(this.uri, this.workingDirectory);
-            return new FileStatus(length, false, 0, 0, modificationTime, filePath);
+            return convertObjectStatsToFileStatus(objectStat);
         } catch (ApiException e) {
             throw new IOException("statObject", e);
         }
     }
 
+    @Nonnull
```

Comment:
Part of `javax.annotation` so it is part of the language - at the IDE level it gives you warnings when you check something for null when it is non-null and vice versa.

Reply:
No it doesn't, it only says that it does.  See https://www.jetbrains.com/help/idea/nullable-and-notnull-annotations.html#notnull for what IntelliJ *really* does.  In practice AFAIU it enforces some IDE-level checks, but the only way that it can work is if you annotate every single method that we call.  Since it isn't checked AFAICS at compile time (just IDE warnings), and  (e.g.) our current API client doesn't provide nonnullable annotations, and Spark clearly doesn't, and I'm not sure what libraries _do_, I'd rather avoid the false appearance of safety.

Won't block on this.  But if you want a typesafe language, it's not going to be called "Java".